src/apps/Console/Console.js

```diff
@@ -148,20 +54,13 @@ function Console({ apps, wrapper }) {
     },
     [toast, apps, wrapper, performIntents]
   )
-  // Handle DAO execution
+
   const handleDaoExec = useCallback(
```

Comment:
We should DRY-ify these three callbacks; they all do the same thing (and note how right now, `dao exec` behaves _slightly_ differently than the other callbacks—it doesn't emit a toast on failures).

Here's a sketch of what I had in mind, to break up the code even more and simplify things:

```js
// console/handlers/install.js
export function installHandler(...) {
  ...
}

// console/handlers/exec.js
export function execHandler(...) {
  ...
}

// Other handlers ...

// console/handlers/index.js
const handlers = new Map([
  ['act', actHandler],
  ['exec', execHandler],
  ['install', installHandler],
])

export default handlers

// console/Console.js
const executeCommand = useCallback(
  async (commandHandler, params) => {
    setLoading(true)
    try {
      const path = await commandHandler(params, { apps, wrapper })
      performIntents([path])
    } catch (error) {
      console.error(error)
      toast('Command execution failed')
    } finally {
      setLoading(false)
    }
  },
  [toast, apps, wrapper, performIntents]
)

const handleSubmit = useCallback(() => {
  const [commandName, ...params] = parsedState
  const handler = handlers[commandName]

  if (typeof handler !== 'function') {
    toast('Unrecognized command')
    handleChange('')
    return
  }

  executeCommand(handler, params)
}, [parsedState, toast, handleChange])
```

Note that the above also changes the handler argument structure slightly, to move the "context"-related variables to an options object (not all commands may need to use them).